### PR TITLE
DBZ-7368: add more documentation for converters in outbox router

### PR DESCRIPTION
**Pull request by enzo-cappa**
Re-structure and add more documentation about setting up the converters in the Outbox Event Router SMT. My objective is make it clear how to properly use JSON and Avro regarding the value converters, the db column types, and the delegate converter configuration.

I'm also adding an extra example of how to setup a delegate converter in Avro format.

Needless to say, this is just a suggestion and is open to any comment or recommendation.

Jira [DBZ-7368](https://issues.redhat.com/browse/DBZ-7368)

### CODE REVIEW

**Review by Naros — APPROVED** [review]
This looks okay from my PoV, I'll defer the final verdict to @roldanbob .

**Comment by roldanbob** [comment]
I've started a review, but was unable to finish today. I'll complete the work tomorrow (1 FEB). 
Please hold off on merging pending the completed review. Thanks.

**Review by mfvitale — APPROVED** [review]

File: documentation/modules/ROOT/pages/transformations/outbox-event-router.adoc
Comment (roldanbob) [inline documentation/modules/ROOT/pages/transformations/outbox-event-router.adoc:231]:
```suggestion
When a {prodname} outbox message represents the `payload` as a JSON String, the resulting Kafka message  escapes the string as in the following example:
```

File: documentation/modules/ROOT/pages/transformations/outbox-event-router.adoc
Comment (roldanbob) [inline documentation/modules/ROOT/pages/transformations/outbox-event-router.adoc:245]:
```suggestion
The outbox event router enables you to expand the message content to "real" JSON, deducing the companion schema from the JSON document. 
The resulting Kafka message is formatted as in the following example:
```

File: documentation/modules/ROOT/pages/transformations/outbox-event-router.adoc
Comment (roldanbob) [inline documentation/modules/ROOT/pages/transformations/outbox-event-router.adoc:258]:
```suggestion
To enable use of the outbox event router transformation, set the xref:outbox-event-router-property-table-expand-json-payload[`table.expand.json.payload`] to true, and use the `JsonConverter` as shown in the following example:
```

File: documentation/modules/ROOT/pages/transformations/outbox-event-router.adoc
Comment (roldanbob) [inline documentation/modules/ROOT/pages/transformations/outbox-event-router.adoc:270]:
```suggestion
// ModuleID: outbox-event-router-using-avro-as-the-payload-format-in-debezium-outbox-messages
```

File: documentation/modules/ROOT/pages/transformations/outbox-event-router.adoc
Comment (roldanbob) [inline documentation/modules/ROOT/pages/transformations/outbox-event-router.adoc:274]:
```suggestion
A common practice is to serialize data as Avro. 
Using Avro can be beneficial for message format governance and for ensuring that outbox event schemas evolve in a backwards-compatible way.
```

File: documentation/modules/ROOT/pages/transformations/outbox-event-router.adoc
Comment (roldanbob) [inline documentation/modules/ROOT/pages/transformations/outbox-event-router.adoc:288]:
```suggestion
By default, the `payload` column value (the Avro data) is the only message value. 
When data is stored in Avro format, the column format must be set to a binary data type, such as `bytea` in PostgreSQL.
```

File: documentation/modules/ROOT/pages/transformations/outbox-event-router.adoc
Comment (roldanbob) [inline documentation/modules/ROOT/pages/transformations/outbox-event-router.adoc:289]:
```suggestion
The value converter for the SMT must also be set to `BinaryDataConverter`, so that it propagates the binary value of the `payload` column as-is into the Kafka message value.
```

File: documentation/modules/ROOT/pages/transformations/outbox-event-router.adoc
Comment (roldanbob) [inline documentation/modules/ROOT/pages/transformations/outbox-event-router.adoc:307]:
```suggestion
The following example illustrates how to use a delegate converter in Avro format with Schema Registry:
```

Comment (roldanbob) [reply documentation/modules/ROOT/pages/transformations/outbox-event-router.adoc:307]: Should the reference to _Schema Registry_ be changed to _Apicurio Registry_?

File: documentation/modules/ROOT/pages/transformations/outbox-event-router.adoc
Comment (roldanbob) [inline documentation/modules/ROOT/pages/transformations/outbox-event-router.adoc:322]:
```suggestion
In the preceding configuration example, because the `AvroConverter` is configured as a delegate converter, third-party libraries are required. 
Information about how to add third-party libraries to the classpath is beyond the scope of this document.
```

**Review by roldanbob — CHANGES_REQUESTED** [review]

File: documentation/modules/ROOT/pages/transformations/outbox-event-router.adoc
Comment (enzo-cappa) [reply documentation/modules/ROOT/pages/transformations/outbox-event-router.adoc:213]: It makes sense to me. This paragraph is mostly based on existing content, so I don't have any strong opinion about it.

File: documentation/modules/ROOT/pages/transformations/outbox-event-router.adoc
Comment (enzo-cappa) [reply documentation/modules/ROOT/pages/transformations/outbox-event-router.adoc:307]: I don't have access to a running instance of Apicurio Registry in order to test the setup and provide an example. But yes, if you think it would be better to document Apicurio, we can do it. Would the example below would still be valid?

**Comment by enzo-cappa** [comment]
Thanks @roldanbob for the thorough review. The result exceeds my original expectations. The only remaining question is about Apicurio Registry vs Schema Registry.

**Comment by roldanbob** [comment]
> Thanks @roldanbob for the thorough review. The result exceeds my original expectations. The only remaining question is about Apicurio Registry vs Schema Registry.

Oh, great. My comment about Apicurio comes from the Red Hat perspective, since I'm familiar with it bc one of my colleagues works on the documentation for it. But I realize that you might have intended a generic reference. Assuming that's the case, and you weren't referring to a specific project, I've modified my original comment to use lowercase 
 `s` and `r` to indicate a generic vs. specific project or product.

File: documentation/modules/ROOT/pages/transformations/outbox-event-router.adoc
Comment (jpechane) [reply documentation/modules/ROOT/pages/transformations/outbox-event-router.adoc:312]: Hi, yes, I'd prefer to have either Apicurio registry example or both. Please check https://github.com/debezium/debezium/blob/main/debezium-connector-postgres/pom.xml#L227 how do we start the simplest Apicurio registry deployment. It can be started easily via `docker run`

Comment (enzo-cappa) [reply documentation/modules/ROOT/pages/transformations/outbox-event-router.adoc]: Added example for apicurio

**Comment by roldanbob** [comment]
@enzo-cappa ICYMI, in my last comment, I identified an extra space on L34. Other than that this LGTM. Thanks!

File: documentation/modules/ROOT/pages/transformations/outbox-event-router.adoc
Comment (enzo-cappa) [inline documentation/modules/ROOT/pages/transformations/outbox-event-router.adoc:234]:
```suggestion
When a {prodname} outbox message represents the `payload` as a JSON String, the resulting Kafka message escapes the string as in the following example:
```

File: documentation/modules/ROOT/pages/transformations/outbox-event-router.adoc
Comment (roldanbob) [inline documentation/modules/ROOT/pages/transformations/outbox-event-router.adoc:275]:
```suggestion
=== Using Apache Avro as the payload format
```

File: documentation/modules/ROOT/pages/transformations/outbox-event-router.adoc
Comment (roldanbob) [inline documentation/modules/ROOT/pages/transformations/outbox-event-router.adoc:277]:
```suggestion
Apache Avro is a common framework for serializing data. 
```

File: documentation/modules/ROOT/pages/transformations/outbox-event-router.adoc
Comment (roldanbob) [inline documentation/modules/ROOT/pages/transformations/outbox-event-router.adoc:272]:
```suggestion
// Title: Using Apache Avro as the payload format in {prodname} outbox messages
```

Comment (roldanbob) [reply documentation/modules/ROOT/pages/transformations/outbox-event-router.adoc:272]: I know that this is pre-existing content, and not directly related to this PR, but I'd appreciate it if you could correct this oversight as part of your update. 🙏🏼

**Review by roldanbob — CHANGES_REQUESTED** [review]
@enzo-cappa I took the time to go over the content once more and identified a couple of small changes that I think we should include. Apologies for not finding these issues in my original review. Thanks!

**Review by roldanbob — APPROVED** [review]
LGTM! Thanks for your efforts and patience!

**Comment by enzo-cappa** [comment]
Thanks for the review, @roldanbob. My part of this PR is ready to be merged whenever you consider it appropriate (I don't have write access).

**Pull request merged**
